pkg/serverless/metrics/enhanced_metrics.go

```diff
@@ -301,6 +304,67 @@ func SendCPUEnhancedMetrics(userCPUOffsetMs, systemCPUOffsetMs float64, tags []s
 	})
 }
 
+func SendNetworkEnhancedMetrics(networkOffsetData *proc.NetworkData, tags []string, demux aggregator.Demultiplexer) {
+	networkData, err := proc.GetNetworkData(proc.ProcNetDevPath)
+	if err != nil {
+		log.Debug("Could not emit network enhanced metrics")
+		return
+	}
+
+	now := float64(time.Now().UnixNano()) / float64(time.Second)
+	GenerateNetworkEnhancedMetrics(GenerateNetworkEnhancedMetricArgs{
+		networkOffsetData.RxBytes,
+		networkData.RxBytes,
+		networkOffsetData.TxBytes,
+		networkData.TxBytes,
+		tags,
+		demux,
+		now,
+	})
+}
+
+type GenerateNetworkEnhancedMetricArgs struct {
+	RxBytesOffset float64
+	RxBytes       float64
+	TxBytesOffset float64
+	TxBytes       float64
+	Tags          []string
+	Demux         aggregator.Demultiplexer
+	Time          float64
+}
+
+func GenerateNetworkEnhancedMetrics(args GenerateNetworkEnhancedMetricArgs) {
+	if strings.ToLower(os.Getenv(enhancedMetricsEnvVar)) == "false" {
+		return
```

Comment:
I would suggest finding a way to halt metric creation earlier. In this case, if the user does not want enhanced metrics, the extension is still calculating the network metrics.

Reply:
Yeah good catch, updated